nomos-blend/network/src/handler.rs

```diff
@@ -244,19 +245,16 @@ where
                 }
                 // If the substream is idle, and if it's time to send a message, send it.
                 Some(OutboundSubstreamState::Idle(stream)) => {
-                    match self.outbound_msgs.pop_front() {
-                        Some(msg) => {
-                            tracing::debug!("Sending message to outbound stream: {:?}", msg);
-                            self.outbound_substream = Some(OutboundSubstreamState::PendingSend(
-                                send_msg(stream, msg).boxed(),
-                            ));
-                        }
-                        None => {
-                            tracing::debug!("Nothing to send to outbound stream");
-                            self.outbound_substream = Some(OutboundSubstreamState::Idle(stream));
-                            self.waker = Some(cx.waker().clone());
-                            return Poll::Pending;
-                        }
+                    if let Some(msg) = self.outbound_msgs.pop_front() {
+                        tracing::debug!("Sending message to outbound stream: {:?}", msg);
+                        self.outbound_substream = Some(OutboundSubstreamState::PendingSend(
+                            send_msg(stream, msg).boxed(),
+                        ));
+                    } else {
+                        tracing::debug!("Nothing to send to outbound stream");
+                        self.outbound_substream = Some(OutboundSubstreamState::Idle(stream));
+                        self.waker = Some(cx.waker().clone());
+                        return Poll::Pending;
```

Comment:
Is there a lint for this? Does it really make sense? :thinking:

Reply:
This is the `clippy::single_match_else` lint. Quoting:

> Just readability – `if let` nests less than a match.

I can also revert this if you guys don't like it, but I don't think it's any worse than the previous way.

Reply:
Clippy is not very happy with wildcard patterns, nor am I to be honest. It is bothersome to consider all possible cases, but I find this to be a strength of Rust as it makes you aware (willingly or not) that stuff is changing, and you are forced to react to them. In this case, we are indeed matching over an option, so there won't be any more branches in the future, making your suggestion of using a `_` equally valid, but I find the suggestion from Clippy on point and a better approach that indeed leads to a layer less of indentation, which never hurts.

Reply:
My point was that I would be ok with replacing `_` patterns but a bit less keen on explicit ones. Anyway, not a blocker, just not sure it's that convenient